File: tmp_node1_sync/rae-core/rae_core/interfaces/embedding.py
"""Abstract embedding provider interface for RAE-core."""

from typing import Protocol, runtime_checkable


@runtime_checkable
class IEmbeddingProvider(Protocol):
    """Abstract interface for embedding providers."""

    async def embed_text(self, text: str) -> list[float]:
        """Generate embedding for text."""
        ...

    async def embed_batch(self, texts: list[str]) -> list[list[float]]:
        """Generate embeddings for multiple texts."""
        ...

    def get_dimension(self) -> int:
        """Get embedding dimension."""
        ...

File: rae-core-rust/src/traits/embedding.rs
use async_trait::async_trait;
use anyhow::Result;

#[async_trait]
pub trait IEmbeddingProvider: Send + Sync {
    async fn embed_text(&self, text: &str) -> Result<Vec<f32>>;
    async fn embed_batch(&self, texts: Vec<String>) -> Result<Vec<Vec<f32>>>;
    fn get_dimension(&self) -> usize;
}
